lib/segment/src/index/query_optimization/condition_converter.rs

```diff
@@ -209,5 +209,27 @@ pub fn get_match_checkers(index: &FieldIndex, cond_match: Match) -> Option<Condi
             }
             _ => None,
         },
+        Match::Except(MatchExcept { except }) => match (except, index) {
+            (AnyVariants::Keywords(list), FieldIndex::KeywordIndex(index)) => {
+                Some(Box::new(move |point_id: PointOffsetType| {
+                    match index.get_values(point_id) {
+                        None => false,
+                        Some(values) => values.iter().any(|k| !list.contains(k)),
+                    }
+                }))
+            }
+            (AnyVariants::Integers(list), FieldIndex::IntMapIndex(index)) => {
+                Some(Box::new(move |point_id: PointOffsetType| {
+                    match index.get_values(point_id) {
+                        None => false,
+                        Some(values) => values.iter().any(|i| !list.contains(i)),
+                    }
+                }))
+            }
+            (_, index) => Some(Box::new(move |point_id: PointOffsetType| {
```

Comment:
`move` is not required here:

```suggestion
            (_, index) => Some(Box::new(|point_id: PointOffsetType| {
```